L1Trigger/L1TMuonEndCap/src/Tree.cc

```
@@ -204,11 +235,19 @@ void Tree::rankVariablesRecursive(Node* node, std::vector<Double_t>& v)
     Node* left = node->getLeftDaughter();
     Node* right = node->getRightDaughter();
 
+    // Terminal nodes don't contribute to error reduction.
     if(left==0 || right==0) return;
 
-    Int_t sv = node->getSplitVariable();
+    Int_t sv =  node->getSplitVariable();
     Double_t er = node->getErrorReduction();
 
+    if(sv == -1)
+    {
+        std::cout << "ERROR: negative split variable for nonterminal node." << std::endl;
```

maybe throw instead of printing?